src/contracts/strategies/StrategyFactory.sol

```diff
@@ -0,0 +1,108 @@
+import "./StrategyFactoryStorage.sol";
+import "./StrategyBase.sol";
+import "../permissions/Pausable.sol";
+
+/**
+ * @title Factory contract for deploying BeaconProxies of a Strategy contract implementation for arbitrary ERC20 tokens
+ *        and automatically adding them to the StrategyWhitelist in EigenLayer.
+ * @author Layr Labs, Inc.
+ * @notice Terms of Service: https://docs.eigenlayer.xyz/overview/terms-of-service
+ * @dev This may not be compatible with non-standard ERC20 tokens. Caution is warranted.
+ */
+contract StrategyFactory is StrategyFactoryStorage, OwnableUpgradeable, Pausable {
+
+    uint8 internal constant PAUSED_NEW_STRATEGIES = 0;
+
+    /// @notice EigenLayer's StrategyManager contract
+    IStrategyManager public immutable strategyManager;
+
+    /// @notice Since this contract is designed to be initializable, the constructor simply sets the immutable variables.
+    constructor(IStrategyManager _strategyManager) {
+        strategyManager = _strategyManager;
+        _disableInitializers();
+    }
+
+    function initialize(
+        address _initialOwner,
+        IPauserRegistry _pauserRegistry,
+        uint256 _initialPausedStatus,
+        IBeacon _strategyBeacon
+    )
+        public virtual initializer
+    {
+        _transferOwnership(_initialOwner);
+        _initializePauser(_pauserRegistry, _initialPausedStatus);
+        _setStrategyBeacon(_strategyBeacon);
+    }
+
+    /**
+     * @notice Deploy a new strategyBeacon contract for the ERC20 token.
+     * @dev A strategy contract must not yet exist for the token.
+     * $dev Immense caution is warranted for non-standard ERC20 tokens, particularly "reentrant" tokens
+     * like those that conform to ERC777.
+     */
+    function deployNewStrategy(IERC20 token) external onlyWhenNotPaused(PAUSED_NEW_STRATEGIES) returns (IStrategy newStrategy) {
+        require(tokenStrategy[token] == IStrategy(address(0)),
+            "StrategyFactory.deployNewStrategy: Strategy already exists for token");
+        IStrategy strategy = IStrategy(address(
+            new BeaconProxy(
+                address(strategyBeacon),
+                abi.encodeWithSelector(StrategyBase.initialize.selector, token, pauserRegistry)
+            )
+        ));
+        _setStrategyForToken(token, strategy);
+        IStrategy[] memory strategiesToWhitelist = new IStrategy[](1);
+        bool[] memory thirdPartyTransfersForbiddenValues = new bool[](1);
+        strategiesToWhitelist[0] = strategy;
+        thirdPartyTransfersForbiddenValues[0] = false;
+        strategyManager.addStrategiesToDepositWhitelist(strategiesToWhitelist, thirdPartyTransfersForbiddenValues);
+        return strategy;
+    }
+
+    /** 
+     * @notice Owner-only function to pass through a call to `StrategyManager.addStrategiesToDepositWhitelist`
+     * @dev Also adds the `strategiesToWhitelist` to the `tokenStrategy` mapping
+     */
+    function whitelistStrategies(
```

Comment:
Seems unnecessary.

Reply:
agreed offline that some passthrough seems prudent; this function is now a pure passthrough (doesn't edit any storage in the StrategyFactory) as of https://github.com/Layr-Labs/eigenlayer-contracts/pull/522/commits/b88cdb5617d8ba51b15b6472f655d8043d1929ee